src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -189,6 +197,103 @@ def add_storage(
             raise NotImplementedError()
         self.storage_types[name] = type_
 
+    def generate_temporaries(
+        self, node_params: list[Sym], defs_state: dace.SDFGState, program_sdfg: dace.SDFG
+    ) -> dict[str, TaskletExpr]:
+        """Create a table of symbols which are used to define array shape, stride and offset for temporaries."""
+        symbol_map: dict[str, TaskletExpr] = {}
+        # The shape of temporary arrays might be defined based on the shape of other input/output fields.
+        # Therefore, here we collect the symbols used to define data-field parameters that are not temporaries.
+        for sym in node_params:
+            if all([sym.id != tmp.id for tmp in self.tmps]) and sym.kind != "Iterator":
+                name_ = str(sym.id)
+                type_ = self.storage_types[name_]
+                assert isinstance(type_, ts.ScalarType)
+                symbol_map[name_] = SymbolExpr(name_, as_dace_type(type_))
+
+        symbol_dtype = dace.int64
+        tmp_symbols: dict[str, TaskletExpr] = {}
+        for tmp in self.tmps:
+            tmp_name = str(tmp.id)
+            assert isinstance(tmp.domain, itir.FunCall)
+            self.node_types.update(itir_typing.infer_all(tmp.domain))
+            domain_ctx = Context(program_sdfg, defs_state, symbol_map)
+            tmp_domain = self._visit_domain(tmp.domain, domain_ctx)
+
+            # First build the FieldType for this temporary array
+            dims: list[Dimension] = []
+            for dim, _ in tmp_domain:
+                dims.append(
+                    Dimension(
+                        value=dim,
+                        kind=(
+                            DimensionKind.VERTICAL
+                            if self.column_axis is not None and self.column_axis.value == dim
+                            else DimensionKind.HORIZONTAL
+                        ),
+                    )
+                )
+            assert isinstance(tmp.dtype, str)
+            type_ = ts.FieldType(dims=dims, dtype=as_scalar_type(tmp.dtype))
+            self.storage_types[tmp_name] = type_
+
+            # N.B.: skip generation of symbolic strides and just let dace assign default strides, for now.
+            # Another option, in the future, is to use symbolic strides and apply auto-tuning or some heuristics
+            # to assign optimal stride values.
+            tmp_shape, _ = new_array_symbols(tmp_name, len(dims))
+            tmp_offset = [
+                dace.symbol(unique_name(f"{tmp_name}_offset{i}")) for i in range(len(dims))
+            ]
+            _, tmp_array = program_sdfg.add_array(
+                tmp_name, tmp_shape, as_dace_type(type_.dtype), offset=tmp_offset, transient=True
+            )
+
+            # Loop through all dimensions to initialize the array parameters for shape and offsets
+            for (_, (begin, end)), offset_sym, shape_sym in zip(
```

Comment:
It looks as if `begin.value` and `end.value` are constant and known during translation time, as the tasklets have no inputs.
So why dou you creates scalars in the first place, you should just create them as symbols (what you later do but you could compute their value here.

Reply:
But you should be able to feed that to symbolic assignements in edges.

Reply:
I remember I tried and it din't work, at that time. I think the problem was in some symbolic expression like `some_symbol - 1`, but I could be wrong. Anyway, it seems to work for now so I will change as you suggest.

Reply:
Oh no, here we go:
```
../../../../../src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py:363: in visit_FencilDefinition
    program_sdfg.validate()
/Users/epaone/repo/dace/dace/sdfg/sdfg.py:2314: in validate
    validate_sdfg(self, references, **context)
/Users/epaone/repo/dace/dace/sdfg/validation.py:264: in validate_sdfg
    validate_control_flow_region(sdfg, sdfg, initialized_transients, symbols, references, **context)
/Users/epaone/repo/dace/dace/sdfg/validation.py:79: in validate_control_flow_region
    undef_syms = set(edge.data.free_symbols) - set(symbols.keys())
/Users/epaone/repo/dace/dace/sdfg/sdfg.py:287: in free_symbols
    return self.used_symbols(all_symbols=True)
/Users/epaone/repo/dace/dace/sdfg/sdfg.py:276: in used_symbols
    rhs_symbols |= set(map(str, dace.symbolic.symbols_in_ast(ast.parse(rhs))))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

source = "SymbolExpr(value='0 - 0', dtype=long long)", filename = '<unknown>'
mode = 'exec'

    def parse(source, filename='<unknown>', mode='exec', *,
              type_comments=False, feature_version=None):
        """
        Parse the source into an AST node.
        Equivalent to compile(source, filename, mode, PyCF_ONLY_AST).
        Pass type_comments=True to get back type comments where the syntax allows.
        """
        flags = PyCF_ONLY_AST
        if type_comments:
            flags |= PyCF_TYPE_COMMENTS
        if isinstance(feature_version, tuple):
            major, minor = feature_version  # Should be a 2-tuple.
            assert major == 3
            feature_version = minor
        elif feature_version is None:
            feature_version = -1
        # Else it should be an int giving the minor version for 3.x.
>       return compile(source, filename, mode, flags,
                       _feature_version=feature_version)
E         File "<unknown>", line 1
E           SymbolExpr(value='0 - 0', dtype=long long)
E                                           ^^^^^^^^^
E       SyntaxError: invalid syntax. Perhaps you forgot a comma?

/opt/homebrew/Cellar/python@3.10/3.10.13_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ast.py:50: SyntaxError
```

Reply:
and different error if I change dtype to `int32`

Reply:
Right, probably my mistake.